deepchem/models/lightning/tests/test_lightning_trainer.py

```diff
@@ -0,0 +1,222 @@
+import pytest
+import deepchem as dc
+import numpy as np
+import os
+try:
+    import torch
+    gpu_available = torch.cuda.is_available() and torch.cuda.device_count() > 0
+except ImportError:
+    gpu_available = False
+
+try:
+    import lightning as L
+    from deepchem.models.lightning.trainer import LightningTorchModel
+    PYTORCH_LIGHTNING_IMPORT_FAILED = False
+except ImportError:
+    PYTORCH_LIGHTNING_IMPORT_FAILED = True
+
+pytestmark = [
+    pytest.mark.skipif(not gpu_available,
+                       reason="No GPU available for testing"),
+    pytest.mark.skipif(PYTORCH_LIGHTNING_IMPORT_FAILED,
+                       reason="PyTorch Lightning is not installed")
+]
+
+
+@pytest.mark.torch
+def test_multitask_classifier_reload_correctness():
+    L.seed_everything(42)
+    tasks, datasets, _ = dc.molnet.load_clintox()
+    _, valid_dataset, _ = datasets
+
+    model = dc.models.MultitaskClassifier(n_tasks=len(tasks),
+                                          n_features=1024,
+                                          layer_sizes=[1000],
+                                          dropouts=0.2,
+                                          learning_rate=0.0001,
+                                          device="cpu",
+                                          batch_size=16)
+
+    trainer = LightningTorchModel(model=model,
+                                  batch_size=16,
+                                  max_epochs=30,
+                                  accelerator="cuda",
+                                  devices=-1,
+                                  log_every_n_steps=1,
+                                  strategy="fsdp",
+                                  fast_dev_run=True)
+
+    trainer.fit(valid_dataset)
+    # get a some 10 weights for assertion
+    weights = trainer.model.model.layers[0].weight[:10].detach().cpu().numpy()
+
+    trainer.save_checkpoint("multitask_classifier.ckpt")
```

Comment:
You should implement a custom save function so the user does not have to do this. Make it something automatic

```
trainer.save(...)
```

We don't have save_checkpoint elsewhere in deepchem. Please make it fit with the standard deepchem API for save/reload